src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/TokenBasedFormattingRule.cs

```diff
@@ -510,7 +510,7 @@ private static SyntaxList<UsingDirectiveSyntax> GetUsings(SyntaxNode node)
             }
 
             // pointer case
-            if ((currentToken.Kind() == SyntaxKind.AsteriskToken && currentToken.Parent is PointerTypeSyntax) ||
+            if ((currentToken.Kind() == SyntaxKind.AsteriskToken && (currentToken.Parent is PointerTypeSyntax || currentToken.Parent?.Kind() == SyntaxKindEx.FunctionPointerType)) ||
                 (previousToken.Kind() == SyntaxKind.AsteriskToken && previousToken.Parent is PrefixUnaryExpressionSyntax))
```

Comment:
can you just make this int multiple checks?  i can't read this anymore :)

Reply:
I could barely read it originally, so figured I couldn't make it worse. ;-) (fixing)